lib/ansible/utils/unsafe_proxy.py

```diff
@@ -369,3 +371,17 @@ def to_unsafe_text(*args, **kwargs):
 
 def _is_unsafe(obj):
     return getattr(obj, '__UNSAFE__', False) is True
+
+
+def _intern(string):
+    """This is a monkey patch for ``sys.intern`` that will strip
+    the unsafe wrapper prior to interning the string.
+
+    This will not exist in future versions.
+    """
+    if isinstance(string, AnsibleUnsafeText):
+        string = string._strip_unsafe()
+    return _sys_intern(string)
+
+
+sys.intern = _intern
```

Comment:
One minor nit here- IIRC some of our unit/sanity tests still do bulk module unload/reload, so for a patch-on-load thing like this, it's probably a good idea to sniff the function to be patched before patching it again to avoid nesting patches in that case, eg:

```suggestion
if type(sys.intern) is type(sys.exc_info):  # builtin functions have a different type than user functions
    sys.intern = _intern
else:
    warnings.warn("skipped sys.intern patch; appears to have already been patched")
```

Reply:
Updated with a slightly different defensive patching.